avl_tree/AVL_Tree.java

```diff
@@ -0,0 +1,294 @@
+import java.util.Scanner;
+
+public class AVL_Tree {
+
+	public static void main(String[] args) {
+		AVL avl = new AVL();
+		int a;
+		Scanner inp = new Scanner(System.in);
+		while (true) {
+			System.out.println("\nTo insert an element Enter 1");
+			System.out.println("To delete an element Enter 2");
+			System.out.println("To print AVL tree preorder Enter 3");
+			System.out.println("To EXIT Enter 4");
+
+			a = inp.nextInt();
+			switch (a) {
+			case 1:
+				int in = inp.nextInt();
+				avl.insert(in);
+				break;
+			case 2:
+				int d = inp.nextInt();
+				avl.del(d);
+				break;
+			case 3:
+				if(avl.isEmpty()){
+					System.out.println("null");
+				}
+				else{
+					avl.prntIn(avl.root);
+					System.out.println(" ");
+				}
+				break;
+			case 4:

```

Comment:
Okay 👍 . But we follow hardcoding the inputs rather than making user input the entries.
Also you should display how the exceptions will work that the DS should throw in case of some errors.
Please Hardcode the input.